gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/RubricPreviewPanel.java

```diff
@@ -0,0 +1,102 @@
+import org.apache.wicket.markup.html.basic.Label;
+import org.apache.wicket.markup.html.WebMarkupContainer;
+import org.apache.wicket.markup.html.form.Form;
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.model.ResourceModel;
+import org.apache.wicket.model.StringResourceModel;
+
+import org.sakaiproject.gradebookng.business.GradebookNgBusinessService;
+import org.sakaiproject.gradebookng.business.model.GbGradeInfo;
+import org.sakaiproject.gradebookng.business.model.GbUser;
+import org.sakaiproject.gradebookng.tool.component.GbAjaxButton;
+import org.sakaiproject.portal.util.PortalUtils;
+import org.sakaiproject.rubrics.logic.RubricsConstants;
+import org.sakaiproject.service.gradebook.shared.Assignment;
+
+public class RubricPreviewPanel extends BasePanel {
+
+    @SpringBean(name = "org.sakaiproject.gradebookng.business.GradebookNgBusinessService")
+    private GradebookNgBusinessService businessService;
+
+    private final ModalWindow window;
+
+    public RubricPreviewPanel(final String id, final IModel<Map<String, Object>> model, final ModalWindow window) {
+        super(id, model);
+        this.window = window;
+    }
+
+    @Override
+    public void onInitialize() {
+        super.onInitialize();
+        final Map<String, Object> modelData = (Map<String, Object>) getDefaultModelObject();
+        final Long assignmentId = (Long) modelData.get("assignmentId");
+        final WebMarkupContainer sakaiRubricPreview = new WebMarkupContainer("sakai-rubric-student");
+        Assignment assignmentNow = businessService.getAssignment(assignmentId);
+        if(assignmentNow!=null && assignmentNow.isExternallyMaintained()){  //this is an externally-maintained item from Assignments
+            sakaiRubricPreview.add(AttributeModifier.append("entity-id", extractAssignmentId(assignmentNow.getExternalId())));  //rubric association needs Assignment id, not Gradebook id
+            sakaiRubricPreview.add(AttributeModifier.append("tool-id", RubricsConstants.RBCS_TOOL_ASSIGNMENT));
+        } else {
+            sakaiRubricPreview.add(AttributeModifier.append("entity-id", assignmentId));
+            sakaiRubricPreview.add(AttributeModifier.append("tool-id", RubricsConstants.RBCS_TOOL_GRADEBOOKNG));
+        }
+        sakaiRubricPreview.add(AttributeModifier.append("token", rubricsService.generateJsonWebToken(RubricsConstants.RBCS_TOOL_GRADEBOOKNG)));
+        sakaiRubricPreview.add(AttributeModifier.append("instructor", "true"));
+        add(sakaiRubricPreview);
+        final GbAjaxButton done = new GbAjaxButton("done") {
+            @Override
+            public void onSubmit(final AjaxRequestTarget target, final Form<?> form) {
+                RubricPreviewPanel.this.window.close(target);
+            }
+        };
+        done.setDefaultFormProcessing(false);
+        add(done);
+        this.window.setInitialWidth(1100);
+        Assignment assignment = businessService.getAssignment(assignmentId);
```

Comment:
Is it necessary to get this Assignment again or could assignmentNow.getName() be used?

Reply:
It can; I've changed it and it works fine. Thanks for the catch.